external/js/cothority/src/network/proto.ts

```diff
@@ -251,15 +218,28 @@ export class ServerIdentity extends Message<ServerIdentity> {
     }
 
     /**
-     * Returns websocket version of this.url if set, otherwise converts the server
-     * address to match the websocket format.
+     * Returns websocket URL to connect to this ServerIdentity.
      * @returns the websocket address
      */
     getWebSocketAddress(): string {
-        if (this.url) {
-            return ServerIdentity.urlToWebsocket(this.url);
+        if (this.url !== undefined && this.url !== "") {
+            const url = new URL(this.url, {});
+            // TODO why not supporting proto ws: & wss: ?
```

Comment:
I guess the todo can be removed

Reply:
Oups, I should have put it in the description.
Still, wouldn't an URL with proto `ws` be more correct than `http`? The first one is clearly the protocol used.

Reply:
Historically it has been `http` so let's keep it that way.